odoo/tools/config.py

```diff
@@ -190,7 +190,9 @@ def _build_cli(self):
         group.add_option("-u", "--update", dest="update", type='comma',  metavar="MODULE,...", my_default=[], file_loadable=False,
                          help="update one or more modules (comma-separated list, use \"all\" for all modules). Requires -d.")
         group.add_option("--without-demo", dest="without_demo", my_default=False, type='without_demo', metavar='BOOL', nargs='?', const=True,
-                         help="use with -i/--init, skip installing fake demonstration data (e.g. Mitchel Admin/Azure Interior)")
+                         help="(DEPRECATED) use with -i/--init, skip installing fake demonstration data (e.g. Mitchel Admin/Azure Interior)")
+        group.add_option("--with-demo", dest="with_demo", action="store_true", my_default=False,
+                         help="Enable loading demo data for modules to be installed. Requires -d and -i. Default is %default")
```

Comment:
@kmagusiak 

I have a question about this for you actually. I was maybe thinking we should actually take the current behavior of the `without-demo` flag.
I believe, with the current state of this PR (mind you I haven't really tested this), if one would add `with_demo` in their config file, they would have no possibility of disabling demo data through flags when executing odoo.

Perhaps you have more hindsight on this?

Reply:
I think current `--without-demo` is required in production database?
If a user really wants to use demo data. (e.g) duplicate database and test some new modules, he should click `Load demo data` Button first which triggers `force_demo` and then install new modules
<img width="414" alt="Screenshot 2024-11-29 at 13 30 59" src="https://github.com/user-attachments/assets/f334c808-bf0c-4457-9962-7e78af92a0a2">

> if one would add with_demo in their config file, they would have no possibility of disabling demo data through flags when executing odoo.

I think it is not useful since `with_demo` means the server is not for production, and we don't need to support the feature. But technically it should be possible. The user can write `env['ir.module.module'].search([]).demo = False` first, and then install new modules

Reply:
We still need a command line to initialize a database for demo purposes.
The sugestion to support both `--with-demo` and `--without-demo` of @xmo-odoo makes sense. This makes this a boolean flag (not a list of modules for which we want or do not want the demo data. We could stop deprecating that command line, but read only with demo from the configuration file.